kube-runtime/src/controller/mod.rs

```diff
@@ -472,7 +472,9 @@ where
                     obj_ref,
                     reason: reschedule_reason,
                 },
-                run_at: reconciler_finished_at + requeue_after,
+                run_at: reconciler_finished_at
+                    .checked_add(requeue_after)
+                    .unwrap_or_else(Instant::now),
```

Comment:
If this overflows, it would be so far in the future that it practically never runs, right? -> It shouldn't be queued for now, but for never?

Reply:
Yeah, maybe. Not sure how to represent that though, nor if it is wise to try to represent this in the scheduler: if people don't want the reconcile to hit they maybe shouldn't return a requeue.

I'm also not sure if there is some other monotonicity reason here that could cause the panic.

Reply:
I actually remembered that I've investigated this before ... [tokio](https://github.com/tokio-rs/tokio/blob/6fcd9c02176bf3cd570bc7de88edaa3b95ea480a/tokio/src/time/instant.rs#L57-L63) handles it this way, so the default could be 30 years in the future.

Reply:
ah nice! yeah `Instant::far_future` is probably a better default here if there is no other way it can fail.

if they told us to use enormous requeue values, then it's kind of expected that we never get around to running it (and we will always override with the smallest instant here for other changes, so should be safe)